docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc

```diff
@@ -640,7 +640,7 @@ Injection of the `JsonWebToken` and `AccessTokenCredential` is supported in both
 
 Quarkus OIDC uses the refresh token to refresh the current ID and access tokens as part of its <<session-management,session management>> process.
 
-[[user-info]]
+[[user-info-b]]
```

Comment:
Hi @rolfedh What is the idea behind `-a`, `-b` suffixes ? It can become tricky to follow/sync across the docs

Reply:
Hi @sberyozkin. Thanks for asking. When the community guides are integrated into a product guide, their IDs occupy the same namespace, leading to duplicate ID errors that cause build failures. The simplest way to address this issue is to directly modify the ID values, as shown here. 

Alternatively, a more universal approach involves defining a unique `:context:` attribute (such as the root filename) at the top of the guide and appending `_{context}` to all ID definitions. This method also requires appending `_{context}` to cross-references (xrefs) that use those IDs within the same guide, and appending the literal value of `_{context}` to xrefs or links that use those IDs outside the guide. While I had used a variant of this more complex approach in the previous iteration of the security guides, I am now aiming to simplify it. I'm open to any other suggestions.

Reply:
Thanks @rolfedh for the explanation,  but I have to admit that now having to add these suffixes upstream to various references looks like a workaround which is a bir hard to make systematic.  If that older approach already worked,  why push this extra convention to the upstream docs ?

Reply:
So grok is that IDs become duplicate because you mix guides into one while we don't. Problem is that:

- not everyone (I dare say "most people") will not remember or know about this convention
- noone in community can know which IDs may become duplicate because they don't have internal knowledge of product docs

Maybe this https://github.com/asciidoctor/asciidoctor/issues/1696#issuecomment-183873652 could be acceptable (adapted to your use case). I cannot decide this, IMHO you need to include here more people interested in docs with merging power like Max, Guillaume etc.

Reply:
>noone in community can know which IDs may become duplicate because they don't have internal knowledge of product docs

I can imagine so.

>Maybe this https://github.com/asciidoctor/asciidoctor/issues/1696#issuecomment-183873652 could be acceptable (adapted to your use case)

How would it look upstream, for example, what would we have to type upstream to replace  `[[user-info]]` ?

Reply:
I would say docs experts no this better than I do, but:

- what this PR does ATM is basically https://docs.asciidoctor.org/asciidoc/latest/attributes/id/#as-an-inline-anchor
- while what we need is https://docs.asciidoctor.org/asciidoc/latest/sections/id-prefix-and-separator/ so that it can be customizable downstream

I'd like Rolfe to find the solution.

Reply:
For example, I am quite not sure how the referencing works, like, do you use substitution in ref? people will mess this up. Let's leave it to Rolfe...

Reply:
When considering best practices for guide creation, it’s important to address whether duplicating content across multiple guides is advisable. Doing so contradicts the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). I recommend consolidating the information in a single location and using cross-references to direct users to that content.

However, let’s set that aside for now and focus on the current situation. We have two options:

1. **Differentiate Conflicting IDs Only When Necessary**: As we've done in this PR, we can address conflicts by differentiating only the IDs that actually conflict. Most users won’t need to be aware of this convention. When new duplicate IDs appear, the documentation team can simply file a PR to resolve them. This method allows us to ignore the majority of IDs that never come into conflict because they don’t appear in the same downstream product guide.

2. **Apply a General Solution**: This method, similar to what is commonly used in Red Hat product documentation, involves setting a `:context:` attribute at the top of the file with the root filename as its value. Then, you append `_{context}` to every ID in the file. Every cross-reference to that ID would then look like `xref:<filename>#<id-value>_<context-value>`. I’m against this solution because it is over-engineered for the scale of the problem and places an unnecessary burden on the community repository. It’s less intuitive and more challenging for new contributors to grasp: even experienced writers frequently make errors with this approach.

Additionally, keep in mind that Red Hat is rolling out a new documentation platform over the next few quarters. I hope this platform will make this issue obsolete and allow us to automate solutions to this problem entirely downstream.

Reply:
Thanks for info!

> I’m against this solution because it is over-engineered for the scale of the problem and places an unnecessary burden on the community repository. It’s less intuitive and more challenging for new contributors to grasp: even experienced writers frequently make errors with this approach.

I don't share this opinion, it feels logical to me, but other suggestions are probably better and I'd like to avoid introducing something just because of Red Hat if little later it becomes unnecessary.

> Additionally, keep in mind that Red Hat is rolling out a new documentation platform over the next few quarters. I hope this platform will make this issue obsolete and allow us to automate solutions to this problem entirely downstream.

okay